sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -219,6 +219,10 @@ protected String inputToSQL(Expression input) {
     }
   }
 
+  protected String inputToCaseWhenSQL(Expression input) {
+    return "CASE WHEN " + inputToSQL(input) + " THEN 1 ELSE 0 END";
```

Comment:
shall we special-case ALWAYS_TRUE and ALWAYS_FALSE? We don't need to generate CASE WHEN but just integer literals.

Reply:
I tried to make this function as generic as possible,
I do check for TRUE FALSE in MsSqlDialect

Reply:
Converting ALWAYS_TRUE and ALWAYS_FALSE to integer literals is also generic as it works anywhere in the context of converting boolean to int.

Reply:
I see, ok. How should we reformat this?
```suggestion
    if (input is literal) => inputToSql(input)
    else =>
    return "CASE WHEN " + inputToSQL(input) + " THEN 1 ELSE 0 END";
```

Reply:
```
if (always true) {
  return "1"
} else if (always false) {
  return "0"
} else {
  return "CASE WHEN ..."
}
```

Reply:
left a comment for this, I think we should leave this as is. it is more intuitive